components/tag-list/tag-list-item-mixin.js

```diff
@@ -71,24 +108,71 @@ export const TagListItemMixin = superclass => class extends superclass {
 	firstUpdated(changedProperties) {
 		super.firstUpdated(changedProperties);
 
-		const container = this.shadowRoot.querySelector('.tag-list-item-container');
+		const container = this.shadowRoot.querySelector('.tag-list-item-content');
 		this.addEventListener('focus', (e) => {
 			// ignore focus events coming from inside the tag content
-			if (e.composedPath()[0] === this) container.focus();
+			if (e.composedPath()[0] !== this) return;
+			const event = new FocusEvent('focus', { bubbles: true, cancelable: true });
+			container.dispatchEvent(event);
+		});
+		this.addEventListener('blur', () => {
+			const event = new FocusEvent('blur', { bubbles: true, cancelable: true });
+			container.dispatchEvent(event);
 		});
-		this.addEventListener('blur', () => container.blur());
+		this.addEventListener('keydown', this._handleKeydown);
+	}
+
+	deleteItem() {
+		this.parentNode.removeChild(this);
+	}
+
+	handleClearItem(e) {
+		if (!this.clearable) return;
+
+		let handleFocus = false;
+		if (e) {
+			const listItemParent = findComposedAncestor(e.composedPath()[0], (node) => node.role === 'listitem');
```

Comment:
Basically this is so that when "Clear All" is clicked, the focus handling isn't triggered. Something that could be an issue is that it sounds like we want to use tags in `d2l-list-item` so it might be best to do this a different way.